packages/core/src/conversation/ConversationService/ConversationService.test.ts

```diff
@@ -154,6 +162,98 @@ describe('ConversationService', () => {
     });
   });
 
+  describe('"handleEpochMismatch"', () => {
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+
+    const createConversation = (epoch: number, conversationId?: string) => {
+      return {
+        group_id: 'group-id',
+        qualified_id: {id: conversationId || 'conversation-id', domain: 'staging.zinfra.io'},
+        protocol: ConversationProtocol.MLS,
+        epoch,
+      } as Conversation;
+    };
+
+    it('tries to re-join when not established yet conversation found', async () => {
+      const [conversationService, {apiClient}] = buildConversationService();
+
+      const mlsConversation = createConversation(1);
+      const mockedDBResponse: Conversation[] = [mlsConversation];
+      jest.spyOn(apiClient.api.conversation, 'getConversationList').mockResolvedValueOnce({found: mockedDBResponse});
+
+      jest.spyOn(conversationService, 'isMLSConversationEstablished').mockResolvedValueOnce(false);
+
+      await conversationService.handleEpochMismatch();
+      expect(conversationService.joinByExternalCommit).toHaveBeenCalledWith(mlsConversation.qualified_id);
+    });
```

Comment:
I think this test is not necessary as it's also covered by the following test `'re-joins multiple not-established conversations'`